createTranslation() function does not work on App dir (NextJS 15) if I don't retrieve params.lang first.

**What version of this package are you using?** next-translate@2.6.2, with next-translate-plugin@3.0.0-canary.5

**What operating system, Node.js, and npm version?** Windows, Node 22.14.0, and npm 10.8.2

**What happened?** 
In one of my page (/app/[lang]/page.js), I have to retrieve the t function to translate messages.
I'm using this code: const { t } = createTranslation();
But it doesn't work, unless I retrieve the language first.

So, I have to do:

```
export default async function HomePage({ params }) {
  const { lang } = await params;
	const { t } = createTranslation();

```
Instead of simply doing:

```
export default async function HomePage() {
	const { t } = createTranslation();

```

The error message displayed is:

```
 ⨯ Error [TypeError]: Cannot read properties of undefined (reading 'localesToIgnore')
    at Module.generateMetadata (webpack-internal:///(rsc)/./app/%5Blang%5D/page.js:44:96)
    at JSON.parse (<anonymous>) {
  environmentName: 'Server',
  digest: '2727303738'
}
```

**What did you expect to happen?** It should work without having to retrieve the language first.

**Are you willing to submit a pull request to fix this bug?** No.

When I try the canary in next-translate, I have the following error when doing npm install:

npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: next-translate-plugin@3.0.0-canary.5
npm error Found: next-translate@3.0.0-canary.5
npm error node_modules/next-translate
npm error   next-translate@"^3.0.0-canary.5" from the root project
npm error
npm error Could not resolve dependency:
npm error peer next-translate@">= 2.4.1" from next-translate-plugin@3.0.0-canary.5
npm error node_modules/next-translate-plugin
npm error   dev next-translate-plugin@"^3.0.0-canary.5" from the root project
npm error
npm error Conflicting peer dependency: next-translate@2.6.2
npm error node_modules/next-translate
npm error   peer next-translate@">= 2.4.1" from next-translate-plugin@3.0.0-canary.5
npm error   node_modules/next-translate-plugin
npm error     dev next-translate-plugin@"^3.0.0-canary.5" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
npm error
npm error

And I can't even start my project.

--- a/src/createTranslation.tsx
+++ b/src/createTranslation.tsx
@@ -19,6 +19,6 @@ export default function createTranslation(defaultNS?: string) {
     return wrapTWithDefaultNs(t, defaultNS)
   }
 
-  const t = isServer() ? getT() : useMemo(getT, [defaultNS])
+  const t = isServer() ? getT() : useMemo(getT, [defaultNS, lang])
   return { t, lang }
 }

